ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,217 @@
+"""
+Utils for checking functions and methods arguments
+"""
+
+# Copyright (C) 2021-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import typing
+from os.path import exists
+
+from numpy import floating
+from omegaconf import DictConfig
+from yaml import safe_load
+
+
+def raise_value_error_if_parameter_has_unexpected_type(
+    parameter, parameter_name, expected_type
+):
+    """Function raises ValueError exception if parameter has unexpected type"""
+    if not isinstance(parameter, expected_type):
+        parameter_type = type(parameter)
+        raise ValueError(
+            f"Unexpected type of '{parameter_name}' parameter, expected: {expected_type}, actual: {parameter_type}"
+        )
+
+
+def check_parameter_type(parameter, parameter_name, expected_type):
+    """Function extracts nested expected types and raises ValueError exception if parameter has unexpected type"""
+    # pylint: disable=W0212
+    if isinstance(expected_type, typing._GenericAlias):  # type: ignore
+        origin_class = expected_type.__dict__.get("__origin__")
```

Comment:
Please, add before this line special check  `if expected_type == typing.Any`
In this case just return. (Note that `Any.__dict__ = {}`)